scripts/generate/public_source_gen.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright (c) F5, Inc.
+# This source code is licensed under the Apache License, Version 2.0 license found in the
+# LICENSE file in the root directory of this source tree.
+
+# shellcheck shell=bash
+# Generate support file from a public git repo
+# Only call it through //go:generate comments in project root to update .gen.go files
+
+set -e
+
+branch="master"
+url=""
+output_path=""
+sub_path=""
+config_path=""
+
+help() {
+    cat << EOF
+Generate support file from a public repo(currently only git).
+Only call me through //go:generate comments in project root to update .gen.go files.
+
+usage: $(basename "$0") [-b|--branch] [-o|--output-path] [-c|--config-path] [-s|--sub-path] [--url] [-h|--help]
+    -h  | --help                Display this message
+    -b  | --branch              Branch to checkout, defaults to "$branch". (optional)
+    -o  | --output-path         The path of the output file. (required)
+    -c  | --config-path         The path of json config file. Normally the json config file should be under nginx-go-crossplane/scripts/generate/configs.
+     The json config will be unmarshed into generator.GenerateConfig(at nginx-go-crossplane/internal/generator.go).
+     For the json config format, please refer to examples under nginx-go-crossplane/scripts/generate/configs,
+     or see generator.GenerateConfig directly. (required)
```

Comment:
Why is this required?

```suggestion
    -c  | --config-path         The path of json config file. (required)
```

Reply:
Just want to let the script simple and have a unique way to configure for `go generate`. If we provide `--match-func-name`, `--directive-map-name`, `--filter`.... here like `./cmd/generate`, there maybe some modules doesn't have json config while some others have. A json config for every module seems more tidy?

Reply:
I think less tidy and annoying if nothing in the config file is actually needed to generate the module code.

Reply:
I modified the params of `generate.sh`, kept them same as `./cmd/generate`. For those don't have `filter` and `override` I put them directly in `//go:generate` commands in `analyze.go` But now there are a lot of params(maybe it's okay? since we indeed have such many params).

Reply:
I'm not sure why we're talking about the number of params in the generate directive. My comment was only that the config file should be optional. If you want to use it to keep line lengths short, you should use it.

Reply:
Get it! Applied: use `--config-path` for all `go:generate` comments in `analyzed.go` to shorten line length, while `--config-path` is optional. Added documents to `./scripts/generate/generate.sh`